app/screens/terms_of_service/terms_of_service.js

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    InteractionManager,
+    ScrollView,
+} from 'react-native';
+import {intlShape} from 'react-intl';
+
+import {getTermsOfService, updateTermsOfServiceStatus} from 'app/actions/views/terms_of_service';
+
+import Loading from 'app/components/loading';
+import Markdown from 'app/components/markdown';
+import ErrorBanner from 'app/components/error_banner';
+import StatusBar from 'app/components/status_bar';
+import {getMarkdownTextStyles, getMarkdownBlockStyles} from 'app/utils/markdown';
+import {changeOpacity, makeStyleSheetFromTheme, setNavigatorStyles} from 'app/utils/theme';
+
+export default class TermsOfService extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            logout: PropTypes.func.isRequired,
+        }),
+        closeButton: PropTypes.object,
+        navigator: PropTypes.object,
+        theme: PropTypes.object,
+    };
+
+    static contextTypes = {
+        intl: intlShape,
+    };
+
+    static defaultProps = {
+        termsEnabled: true,
+    };
+
+    leftButton = {
+        id: 'close-terms-of-service',
+    };
+
+    rightButton = {
+        id: 'accept-terms-of-service',
+        showAsAction: 'always',
+    };
+
+    constructor(props, context) {
+        super(props);
+
+        this.state = {
+            error: null,
+            loading: true,
+            serverError: null,
+            termsId: '',
+            termsText: '',
+        };
+
+        this.rightButton.title = context.intl.formatMessage({id: 'terms_of_service.agreeButton', defaultMessage: 'I Agree'});
+        this.leftButton.icon = props.closeButton;
+
+        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+        this.setNavigatorButtons();
```

Comment:
this function will set the left and right buttons as enabled, thus the user could potentially agree to the terms while the terms are being loaded, is that expected?

Reply:
Good point. Will disable the buttons when the terms are being loaded.